diff --git a/CMakeLists.txt b/CMakeLists.txt
index 72cbc45..73a22cd 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -3,7 +3,9 @@ cmake_minimum_required(VERSION 2.8.11)
 set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_SOURCE_DIR}/cmake/modules")
 
 # Build options
-option(WITH_DOC "Build documentation (requires Doxygen)" ON)
+option(WITH_DOC "Build documentation (requires Doxygen)" OFF)
+
+set(CMAKE_INCLUDE_CURRENT_DIR ON)
 
 # Versions
 ## Package version
@@ -107,7 +109,7 @@ install(FILES ${CMAKE_BINARY_DIR}/dbusmenu-${QT_SUFFIX}.pc
     )
 
 add_subdirectory(src)
-add_subdirectory(tests)
+#add_subdirectory(tests)
 add_subdirectory(tools)
 
 if(WITH_DOC)
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 2c4a4e8..9b1a3b6 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -81,7 +81,7 @@ configure_file(dbusmenu_version.h.in
     ${CMAKE_CURRENT_BINARY_DIR}/dbusmenu_version.h
     )
 
-add_library(dbusmenu-${QT_SUFFIX} SHARED ${dbusmenu_qt_SRCS})
+add_library(dbusmenu-${QT_SUFFIX} ${dbusmenu_qt_SRCS})
 set_target_properties(dbusmenu-${QT_SUFFIX} PROPERTIES
     VERSION ${dbusmenu_qt_lib_VERSION}
     SOVERSION ${dbusmenu_qt_lib_SOVERSION}
@@ -110,6 +110,7 @@ target_include_directories(dbusmenu-${QT_SUFFIX}
 install(TARGETS dbusmenu-${QT_SUFFIX}
     EXPORT dbusmenu-${QT_SUFFIX}-targets
     LIBRARY DESTINATION ${LIB_DESTINATION}
+    ARCHIVE DESTINATION ${LIB_DESTINATION}
     RUNTIME DESTINATION bin
     )
 
