diff --git a/autotests/CMakeLists.txt b/autotests/CMakeLists.txt
index ef067fd..f3258a5 100644
--- a/autotests/CMakeLists.txt
+++ b/autotests/CMakeLists.txt
@@ -1,7 +1,12 @@
 
 remove_definitions(-DQT_NO_CAST_FROM_ASCII)
 
-find_package(Qt5Test ${REQUIRED_QT_VERSION} REQUIRED NO_MODULE)
+find_package(Qt5Test ${REQUIRED_QT_VERSION} CONFIG QUIET NO_MODULE)
+
+if(NOT Qt5Test_FOUND)
+    message(STATUS "Qt5Test not found, autotests will not be built.")
+    return()
+endif()
 
 include(ECMAddTests)
 
