diff --git a/autotests/CMakeLists.txt b/autotests/CMakeLists.txt
index ba10d17..9c753d0 100644
--- a/autotests/CMakeLists.txt
+++ b/autotests/CMakeLists.txt
@@ -1,5 +1,9 @@
+find_package(Qt5Test ${REQUIRED_QT_VERSION} CONFIG QUIET)
 
-find_package(Qt5 ${REQUIRED_QT_VERSION} CONFIG REQUIRED Test)
+if(NOT Qt5Test_FOUND)
+    message(STATUS "Qt5Test not found, autotests will not be built.")
+    return()
+endif()
 
 include(ECMAddTests)
 
diff --git a/tests/CMakeLists.txt b/tests/CMakeLists.txt
index 8e664e2..405f69c 100644
--- a/tests/CMakeLists.txt
+++ b/tests/CMakeLists.txt
@@ -1,7 +1,12 @@
 
 include(ECMMarkAsTest)
 
-find_package(Qt5 ${REQUIRED_QT_VERSION} CONFIG REQUIRED Test)
+find_package(Qt5Test ${REQUIRED_QT_VERSION} CONFIG QUIET)
+
+if(NOT Qt5Test_FOUND)
+    message(STATUS "Qt5Test not found, autotests will not be built.")
+    return()
+endif()
 
 macro(kiconthemes_executable_tests)
   foreach(_testname ${ARGN})
