diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2fefcd0..a1268f2 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -73,7 +73,7 @@ remove_definitions(-DQT_NO_CAST_FROM_BYTEARRAY)
 add_definitions(-DQT_NO_URL_CAST_FROM_STRING)
 
 set(REQUIRED_QT_VERSION 5.7.0)
-find_package(Qt5 ${REQUIRED_QT_VERSION} CONFIG REQUIRED Widgets DBus Network Concurrent Xml Test)
+find_package(Qt5 ${REQUIRED_QT_VERSION} CONFIG REQUIRED Widgets DBus Network Concurrent Xml)
 
 find_package(GSSAPI)
 set_package_properties(GSSAPI PROPERTIES DESCRIPTION "Allows KIO to make use of certain HTTP authentication services"
diff --git a/autotests/CMakeLists.txt b/autotests/CMakeLists.txt
index 791eb78..16c53f9 100644
--- a/autotests/CMakeLists.txt
+++ b/autotests/CMakeLists.txt
@@ -3,6 +3,14 @@ if(POLICY CMP0028)
 endif()
 remove_definitions(-DQT_NO_CAST_FROM_ASCII)
 
+find_package(Qt5Test ${REQUIRED_QT_VERSION} CONFIG QUIET)
+
+if(NOT Qt5Test_FOUND)
+    message(STATUS "Qt5Test not found, autotests will not be built.")
+    return()
+endif()
+
+
 include(ECMAddTests)
 
 add_subdirectory(http)
diff --git a/src/ioslaves/trash/tests/CMakeLists.txt b/src/ioslaves/trash/tests/CMakeLists.txt
index f913892..76a4d7f 100644
--- a/src/ioslaves/trash/tests/CMakeLists.txt
+++ b/src/ioslaves/trash/tests/CMakeLists.txt
@@ -1,5 +1,12 @@
 include_directories( ${CMAKE_CURRENT_SOURCE_DIR}/.. )
 
+find_package(Qt5Test ${REQUIRED_QT_VERSION} CONFIG QUIET)
+
+if(NOT Qt5Test_FOUND)
+    message(STATUS "Qt5Test not found, autotests will not be built.")
+    return()
+endif()
+
 ########### next target ###############
 
 set(testtrash_SRCS
diff --git a/src/kpasswdserver/autotests/CMakeLists.txt b/src/kpasswdserver/autotests/CMakeLists.txt
index 3c7e637..6183518 100644
--- a/src/kpasswdserver/autotests/CMakeLists.txt
+++ b/src/kpasswdserver/autotests/CMakeLists.txt
@@ -1,5 +1,12 @@
 include_directories( ${CMAKE_CURRENT_SOURCE_DIR}/.. ${CMAKE_CURRENT_BINARY_DIR}/.. )
 
+find_package(Qt5Test ${REQUIRED_QT_VERSION} CONFIG QUIET)
+
+if(NOT Qt5Test_FOUND)
+    message(STATUS "Qt5Test not found, autotests will not be built.")
+    return()
+endif()
+
 set (kpasswdserver_test_SRCS
  kpasswdservertest.cpp
  ../kpasswdserver.cpp
diff --git a/tests/CMakeLists.txt b/tests/CMakeLists.txt
index bc94a4a..9ba204b 100644
--- a/tests/CMakeLists.txt
+++ b/tests/CMakeLists.txt
@@ -4,6 +4,13 @@ endif()
 
 include(ECMMarkAsTest)
 
+find_package(Qt5Test ${REQUIRED_QT_VERSION} CONFIG QUIET)
+
+if(NOT Qt5Test_FOUND)
+    message(STATUS "Qt5Test not found, autotests will not be built.")
+    return()
+endif()
+
 find_package(KF5XmlGui ${KF5_DEP_VERSION} REQUIRED)
 
 macro(KIOWIDGETS_EXECUTABLE_TESTS)
