diff --git a/autotests/CMakeLists.txt b/autotests/CMakeLists.txt
index dc5af5e..72ac2a2 100644
--- a/autotests/CMakeLists.txt
+++ b/autotests/CMakeLists.txt
@@ -1,5 +1,11 @@
+find_package(Qt5Test ${REQUIRED_QT_VERSION} CONFIG QUIET)
 
-find_package(Qt5 ${REQUIRED_QT_VERSION} CONFIG REQUIRED Xml Test)
+if(NOT Qt5Test_FOUND)
+    message(STATUS "Qt5Test not found, autotests will not be built.")
+    return()
+endif()
+
+find_package(Qt5 ${REQUIRED_QT_VERSION} CONFIG REQUIRED Xml)
 
 remove_definitions(-DQT_NO_CAST_FROM_BYTEARRAY)
 remove_definitions(-DQT_NO_CAST_FROM_ASCII)
diff --git a/src/ksendbugmail/smtp.cpp b/src/ksendbugmail/smtp.cpp
index d26ecf4..aee7693 100644
--- a/src/ksendbugmail/smtp.cpp
+++ b/src/ksendbugmail/smtp.cpp
@@ -173,6 +173,7 @@ void SMTP::connectTimerTick(void)
     sock = nullptr;
 
     // qDebug() << "connecting to " << serverHost << ":" << hostPort << " ..... ";
+
     sock = new QSslSocket(this);
     sock->connectToHost(serverHost, hostPort);
 
diff --git a/tests/CMakeLists.txt b/tests/CMakeLists.txt
index 3a5a42f..a1e1a26 100644
--- a/tests/CMakeLists.txt
+++ b/tests/CMakeLists.txt
@@ -1,7 +1,12 @@
 
 include(ECMMarkAsTest)
 
-find_package(Qt5 ${REQUIRED_QT_VERSION} CONFIG REQUIRED Test)
+find_package(Qt5Test ${REQUIRED_QT_VERSION} CONFIG QUIET)
+
+if(NOT Qt5Test_FOUND)
+    message(STATUS "Qt5Test not found, autotests will not be built.")
+    return()
+endif()
 
 macro(xmlgui_executable_tests)
   foreach(_testname ${ARGN})
