diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6773145c..bcfffc7e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -96,6 +96,8 @@ FIND_PACKAGE(Qt5 ${QT_MIN_VERSION} CONFIG REQUIRED
   PrintSupport
   QuickControls2
   Quick
+  Concurrent
+  X11Extras
 )
 
 #FIND_PACKAGE(Qt5TextToSpeech ${QT_MIN_VERSION} QUIET)
@@ -118,6 +120,9 @@ FIND_PACKAGE(KF5 "${KF5_DEP_VERSION}" REQUIRED COMPONENTS
   XmlGui
   Notifications
   IconThemes
+  ItemViews
+  KIO
+  Archive
   Crash
   NotifyConfig
   GlobalAccel
diff --git a/src/actioncollection.cpp b/src/actioncollection.cpp
index 0ca68a3f..069ca52f 100644
--- a/src/actioncollection.cpp
+++ b/src/actioncollection.cpp
@@ -29,11 +29,11 @@
 #include <KLocalizedString>
 #include <kmessagebox.h>
 #include <KStandardAction>
-#include <KShortcutsDialog>
+//#include <KShortcutsDialog>
 #include <KActionCollection>
 #include <KEditToolBar>
 #include <KSharedConfig>
-#include <KNotifyConfigWidget>
+//#include <KNotifyConfigWidget>
 #include <KGlobalAccel>
 
 //Ring
@@ -336,7 +336,7 @@ void ActionCollection::configureRing()
 ///Display the shortcuts dialog
 void ActionCollection::showShortCutEditor()
 {
-   KShortcutsDialog::configure( RingApplication::instance()->mainWindow()->actionCollection() );
+   //KShortcutsDialog::configure( RingApplication::instance()->mainWindow()->actionCollection() );
 }
 
 ///Display the shortcuts dialog
@@ -348,7 +348,7 @@ void ActionCollection::showDirectory()
 ///Display the notification manager
 void ActionCollection::showNotificationEditor()
 {
-   KNotifyConfigWidget::configure(RingApplication::instance()->mainWindow(), QStringLiteral("ring-kde"));
+//   KNotifyConfigWidget::configure(RingApplication::instance()->mainWindow(), QStringLiteral("ring-kde"));
 }
 
 ///Show the toolbar editor
diff --git a/src/main.cpp b/src/main.cpp
index 98a023ea..87e0c3a4 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -1,3 +1,20 @@
+#include <QtPlugin>
+Q_IMPORT_PLUGIN(QXcbIntegrationPlugin)
+Q_IMPORT_PLUGIN(QtQuick2Plugin)
+Q_IMPORT_PLUGIN(QtQuickControls2Plugin)
+Q_IMPORT_PLUGIN(QtQuick2WindowPlugin)
+Q_IMPORT_PLUGIN(QEvdevKeyboardPlugin)
+Q_IMPORT_PLUGIN(QEvdevMousePlugin)
+Q_IMPORT_PLUGIN(QtQuickLayoutsPlugin)
+Q_IMPORT_PLUGIN(QtQuickTemplates2Plugin)
+Q_IMPORT_PLUGIN(QtQuickControls1Plugin)
+Q_IMPORT_PLUGIN(QJpegPlugin)
+Q_IMPORT_PLUGIN(QSvgPlugin)
+Q_IMPORT_PLUGIN(QSvgIconPlugin)
+Q_IMPORT_PLUGIN(QXcbGlxIntegrationPlugin)
+Q_IMPORT_PLUGIN(QtGraphicalEffectsPlugin)
+Q_IMPORT_PLUGIN(QtGraphicalEffectsPrivatePlugin)
+Q_IMPORT_PLUGIN(QtQmlModelsPlugin)
 /***************************************************************************
  *   Copyright (C) 2009-2015 by Savoir-Faire Linux                         *
  *   Author : Jérémy Quentin <jeremy.quentin@savoirfairelinux.com>         *
