diff --git a/contrib/src/ffmpeg/rules.mak b/contrib/src/ffmpeg/rules.mak
index 5e62a77..262ada1 100644
--- a/contrib/src/ffmpeg/rules.mak
+++ b/contrib/src/ffmpeg/rules.mak
@@ -119,13 +119,8 @@ FFMPEGCONF += \
 	--target-os=linux \
 	--enable-indev=v4l2 \
 	--enable-indev=xcbgrab \
-	--enable-vdpau \
-	--enable-hwaccel=h264_vdpau \
-	--enable-hwaccel=mpeg4_vdpau \
-	--enable-vaapi \
-	--enable-hwaccel=h264_vaapi \
-	--enable-hwaccel=mpeg4_vaapi \
-	--enable-hwaccel=h263_vaapi
+	--disable-vdpau \
+	--disable-vaapi
 endif
 endif
 
