diff --git a/contrib/src/ffmpeg/rules.mak b/contrib/src/ffmpeg/rules.mak
index 119ffb6a3..befbaa8fc 100644
--- a/contrib/src/ffmpeg/rules.mak
+++ b/contrib/src/ffmpeg/rules.mak
@@ -19,7 +19,7 @@ FFMPEGCONF += \
 	--enable-zlib \
 	--enable-gpl \
 	--enable-swscale \
-	--enable-protocols \
+	--disable-protocols \
 	--enable-bsfs \
 	--disable-filters \
 	--disable-programs \
@@ -27,8 +27,150 @@ FFMPEGCONF += \
 
 #enable muxers/demuxers
 FFMPEGCONF += \
-	--enable-demuxers \
-	--enable-muxers
+	--disable-demuxers \
+	--disable-muxers
+
+FFMPEGCONF += \
+	--enable-muxer=ac3 \
+	--enable-muxer=avi \
+	--enable-muxer=avm2 \
+	--enable-muxer=bit \
+	--enable-muxer=crc \
+	--enable-muxer=dv \
+	--enable-muxer=fifo \
+	--enable-muxer=g722 \
+	--enable-muxer=g723_1 \
+	--enable-muxer=g726 \
+	--enable-muxer=g726le \
+	--enable-muxer=gsm \
+	--enable-muxer=h263 \
+	--enable-muxer=h264 \
+	--enable-muxer=hash \
+	--enable-muxer=hevc \
+	--enable-muxer=m4v \
+	--enable-muxer=matroska \
+	--enable-muxer=matroska_audio \
+	--enable-muxer=md5 \
+	--enable-muxer=mjpeg \
+	--enable-muxer=mp2 \
+	--enable-muxer=mp3 \
+	--enable-muxer=mp4 \
+	--enable-muxer=mpeg2dvd \
+	--enable-muxer=mpeg2svcd \
+	--enable-muxer=mpeg2video \
+	--enable-muxer=mpeg2vob \
+	--enable-muxer=mpegts \
+	--enable-muxer=mpjpeg \
+	--enable-muxer=oga \
+	--enable-muxer=ogg \
+	--enable-muxer=ogv \
+	--enable-muxer=opus \
+	--enable-muxer=pcm_alaw \
+	--enable-muxer=pcm_f32be \
+	--enable-muxer=pcm_f32le \
+	--enable-muxer=pcm_f64be \
+	--enable-muxer=pcm_f64le \
+	--enable-muxer=pcm_mulaw \
+	--enable-muxer=pcm_s16be \
+	--enable-muxer=pcm_s16le \
+	--enable-muxer=pcm_s24be \
+	--enable-muxer=pcm_s24le \
+	--enable-muxer=pcm_s32be \
+	--enable-muxer=pcm_s32le \
+	--enable-muxer=pcm_s8 \
+	--enable-muxer=pcm_u16be \
+	--enable-muxer=pcm_u16le \
+	--enable-muxer=pcm_u24be \
+	--enable-muxer=pcm_u24le \
+	--enable-muxer=pcm_u32be \
+	--enable-muxer=pcm_u32le \
+	--enable-muxer=pcm_u8 \
+	--enable-muxer=rawvideo \
+	--enable-muxer=rtp \
+	--enable-muxer=rtp_mpegts \
+	--enable-muxer=rtsp \
+	--enable-muxer=smjpeg \
+	--enable-muxer=smoothstreaming \
+	--enable-muxer=sox \
+	--enable-muxer=spdif \
+	--enable-muxer=spx \
+	--enable-muxer=wav \
+	--enable-muxer=webm \
+	--enable-muxer=webm_chunk \
+	--enable-muxer=webm_dash_manifest \
+	--enable-muxer=yuv4mpegpipe \
+	--enable-muxer=udp \
+	--enable-muxer=udplite \
+	--enable-muxer=unix \
+	--enable-demuxer=aac \
+	--enable-demuxer=sdp \
+	--enable-demuxer=ac3 \
+	--enable-demuxer=avi \
+	--enable-demuxer=dv \
+	--enable-demuxer=flac \
+	--enable-demuxer=g722 \
+	--enable-demuxer=g723_1 \
+	--enable-demuxer=g726 \
+	--enable-demuxer=g726le \
+	--enable-demuxer=g729 \
+	--enable-demuxer=gsm \
+	--enable-demuxer=h263 \
+	--enable-demuxer=h264 \
+	--enable-demuxer=hevc \
+	--enable-demuxer=m4v \
+	--enable-demuxer=matroska \
+	--enable-demuxer=mjpeg \
+	--enable-demuxer=mp3 \
+	--enable-demuxer=mpegps \
+	--enable-demuxer=mpegts \
+	--enable-demuxer=mpegtsraw \
+	--enable-demuxer=mpegvideo \
+	--enable-demuxer=mpjpeg \
+	--enable-demuxer=ogg \
+	--enable-demuxer=pcm_alaw \
+	--enable-demuxer=pcm_f32be \
+	--enable-demuxer=pcm_f32le \
+	--enable-demuxer=pcm_f64be \
+	--enable-demuxer=pcm_f64le \
+	--enable-demuxer=pcm_mulaw \
+	--enable-demuxer=pcm_s16be \
+	--enable-demuxer=pcm_s16le \
+	--enable-demuxer=pcm_s24be \
+	--enable-demuxer=pcm_s24le \
+	--enable-demuxer=pcm_s32be \
+	--enable-demuxer=pcm_s32le \
+	--enable-demuxer=pcm_s8 \
+	--enable-demuxer=pcm_u16be \
+	--enable-demuxer=pcm_u16le \
+	--enable-demuxer=pcm_u24be \
+	--enable-demuxer=pcm_u24le \
+	--enable-demuxer=pcm_u32be \
+	--enable-demuxer=pcm_u32le \
+	--enable-demuxer=pcm_u8 \
+	--enable-demuxer=rawvideo \
+	--enable-demuxer=sox \
+	--enable-demuxer=spdif \
+	--enable-demuxer=wav \
+	--enable-demuxer=webm_dash_manif \
+	--enable-demuxer=yuv4mpegpipe \
+	--enable-protocol=async \
+	--enable-protocol=cache \
+	--enable-protocol=concat \
+	--enable-protocol=crypto \
+	--enable-protocol=data \
+	--enable-protocol=ffrtmphttp \
+	--enable-protocol=file \
+	--enable-protocol=http \
+	--enable-protocol=md5 \
+	--enable-protocol=pipe \
+	--enable-protocol=rtmp \
+	--enable-protocol=rtmpt \
+	--enable-protocol=rtp \
+	--enable-protocol=srtp \
+	--enable-protocol=tcp \
+	--enable-protocol=udp \
+	--enable-protocol=udplite
+
 
 #enable parsers
 FFMPEGCONF += \
@@ -68,7 +210,8 @@ FFMPEGCONF += \
 	--enable-encoder=libspeex \
 	--enable-decoder=libspeex \
 	--enable-encoder=libopus \
-	--enable-decoder=libopus
+	--enable-decoder=libopus \
+        --disable-libopenjpeg
 
 # decoders for ringtones and audio streaming
 FFMPEGCONF += \
@@ -104,18 +247,18 @@ FFMPEGCONF += \
 
 #encoders/decoders for images
 FFMPEGCONF += \
-	--enable-encoder=gif \
-	--enable-decoder=gif \
-	--enable-encoder=jpegls \
-	--enable-decoder=jpegls \
-	--enable-encoder=ljpeg \
-	--enable-decoder=jpeg2000 \
-	--enable-encoder=png \
-	--enable-decoder=png \
-	--enable-encoder=bmp \
-	--enable-decoder=bmp \
-	--enable-encoder=tiff \
-	--enable-decoder=tiff
+	--disable-encoder=gif \
+	--disable-decoder=gif \
+	--disable-encoder=jpegls \
+	--disable-decoder=jpegls \
+	--disable-encoder=ljpeg \
+	--disable-decoder=jpeg2000 \
+	--disable-encoder=png \
+	--disable-decoder=png \
+	--disable-encoder=bmp \
+	--disable-decoder=bmp \
+	--disable-encoder=tiff \
+	--disable-decoder=tiff
 
 #filters
 FFMPEGCONF += \
@@ -165,15 +308,8 @@ FFMPEGCONF += \
 	--target-os=linux \
 	--enable-indev=v4l2 \
 	--enable-indev=xcbgrab \
-	--enable-vdpau \
-	--enable-hwaccel=h264_vdpau \
-	--enable-hwaccel=mpeg4_vdpau \
-	--enable-vaapi \
-	--enable-hwaccel=h264_vaapi \
-	--enable-hwaccel=mpeg4_vaapi \
-	--enable-hwaccel=h263_vaapi \
-	--enable-hwaccel=vp8_vaapi \
-	--enable-hwaccel=mjpeg_vaapi
+        --disable-vdpau \
+        --disable-vaapi
 endif
 endif
 
@@ -257,6 +393,7 @@ ffmpeg: ffmpeg-$(FFMPEG_HASH).tar.gz
 	cd $< && $(HOSTVARS) ./configure \
 		--extra-cflags="$(CFLAGS)" \
 		--extra-ldflags="$(LDFLAGS)" $(FFMPEGCONF) \
-		--prefix="$(PREFIX)" --enable-static --disable-shared
+		--prefix="$(PREFIX)" --enable-static --disable-shared \
+                --enable-small
 	cd $< && $(MAKE) install-libs install-headers
 	touch $@
diff --git a/contrib/src/jsoncpp/rules.mak b/contrib/src/jsoncpp/rules.mak
index ff18b88fc..e19ae86fe 100644
--- a/contrib/src/jsoncpp/rules.mak
+++ b/contrib/src/jsoncpp/rules.mak
@@ -10,7 +10,15 @@ endif
 
 JSONCPP_CMAKECONF := -DBUILD_STATIC_LIBS:BOOL=ON \
                      -DBUILD_SHARED_LIBS:BOOL=OFF \
-                     -DJSONCPP_WITH_TESTS:BOOL=OFF
+                     -DJSONCPP_WITH_TESTS:BOOL=OFF \
+                     -DCMAKE_CXX_FLAGS="$(CXXFLAGS)" \
+                     -DCMAKE_CXX_FLAGS_RELEASE="$(CXXFLAGS)" \
+                     -DCMAKE_C_FLAGS="$(CXXFLAGS)" \
+                     -DCMAKE_C_FLAGS_RELEASE="$(CXXFLAGS)" \
+                     -DCMAKE_AR=/usr/bin/gcc-ar \
+                     -DCMAKE_CXX_ARCHIVE_CREATE="<CMAKE_AR> qcs <TARGET> <LINK_FLAGS> <OBJECTS>" \
+                     -DCMAKE_CXX_ARCHIVE_FINISH=true
+
 
 $(TARBALLS)/jsoncpp-$(JSONCPP_VERSION).tar.gz:
 	$(call download,$(JSONCPP_URL))
@@ -22,6 +30,6 @@ jsoncpp: jsoncpp-$(JSONCPP_VERSION).tar.gz .sum-jsoncpp
 	$(MOVE)
 
 .jsoncpp: jsoncpp toolchain.cmake
-	cd $< && $(HOSTVARS) $(CMAKE) ${JSONCPP_CMAKECONF}
-	cd $< && $(MAKE) install
+	cd $< && $(HOSTVARS) $(CMAKE) . ${JSONCPP_CMAKECONF}
+	cd $< && $(MAKE) install VERBOSE=1
 	touch $@
diff --git a/contrib/src/msgpack/rules.mak b/contrib/src/msgpack/rules.mak
index 2c0479271..be73ee822 100644
--- a/contrib/src/msgpack/rules.mak
+++ b/contrib/src/msgpack/rules.mak
@@ -10,7 +10,15 @@ endif
 MSGPACK_CMAKECONF := -DMSGPACK_CXX11=ON \
 		-DMSGPACK_BUILD_EXAMPLES=OFF \
 		-DMSGPACK_ENABLE_SHARED=OFF \
-		-DCMAKE_INSTALL_LIBDIR=lib
+		-DCMAKE_INSTALL_LIBDIR=lib \
+                -DCMAKE_CXX_FLAGS="$(CXXFLAGS)" \
+                -DCMAKE_CXX_FLAGS_RELEASE="$(CXXFLAGS)" \
+                -DCMAKE_C_FLAGS="$(CXXFLAGS)" \
+                -DCMAKE_C_FLAGS_RELEASE="$(CXXFLAGS)" \
+                -DCMAKE_AR=/usr/bin/gcc-ar \
+                -DCMAKE_CXX_ARCHIVE_CREATE="<CMAKE_AR> qcs <TARGET> <LINK_FLAGS> <OBJECTS>" \
+                -DCMAKE_CXX_ARCHIVE_FINISH=true
+
 
 $(TARBALLS)/msgpack-c-$(MSGPACK_VERSION).tar.gz:
 	$(call download,$(MSGPACK_URL))
@@ -23,5 +31,5 @@ msgpack: msgpack-c-$(MSGPACK_VERSION).tar.gz
 
 .msgpack: msgpack toolchain.cmake .sum-msgpack
 	cd $< && $(HOSTVARS) $(CMAKE) . $(MSGPACK_CMAKECONF)
-	cd $< && $(MAKE) install
+	cd $< && $(MAKE) install VERBOSE=1
 	touch $@
diff --git a/contrib/src/restbed/rules.mak b/contrib/src/restbed/rules.mak
index a8c517bc5..ff2295bbb 100644
--- a/contrib/src/restbed/rules.mak
+++ b/contrib/src/restbed/rules.mak
@@ -39,7 +39,13 @@ RESTBED_CONF = -DBUILD_TESTS=NO \
 			-DBUILD_SHARED=NO \
 			-DCMAKE_INCLUDE_PATH=$(PREFIX)/include \
 			-DCMAKE_INSTALL_PREFIX=$(PREFIX) \
-			-DCMAKE_INSTALL_LIBDIR=lib
+			-DCMAKE_INSTALL_LIBDIR=lib \
+			-DCMAKE_CXX_FLAGS="$(CXXFLAGS)" \
+			-DCMAKE_CXX_FLAGS_RELEASE="$(CXXFLAGS)" \
+			-DCMAKE_AR=/usr/bin/gcc-ar \
+			-DCMAKE_CXX_ARCHIVE_CREATE="<CMAKE_AR> qcs <TARGET> <LINK_FLAGS> <OBJECTS>" \
+			-DCMAKE_CXX_ARCHIVE_FINISH=true
+
 
 restbed: restbed-$(RESTBED_VERSION).tar.gz .sum-restbed
 	$(UNPACK)
@@ -47,7 +53,7 @@ restbed: restbed-$(RESTBED_VERSION).tar.gz .sum-restbed
 
 .restbed: restbed toolchain.cmake
 	cd $< && $(HOSTVARS) $(CMAKE) $(RESTBED_CONF) .
-	cd $< && $(MAKE) install
+	cd $< && $(MAKE) install VERBOSE=1
 	touch $@
 
 .sum-restbed: restbed-$(RESTBED_VERSION).tar.gz
diff --git a/contrib/src/vpx/rules.mak b/contrib/src/vpx/rules.mak
index acaf76e07..c97fdbbd4 100644
--- a/contrib/src/vpx/rules.mak
+++ b/contrib/src/vpx/rules.mak
@@ -87,7 +87,8 @@ VPX_CONF := \
 	--disable-install-docs \
 	--enable-realtime-only \
 	--enable-error-concealment \
-	--disable-webm-io
+	--disable-webm-io \
+	--disable-avx --disable-avx512
 
 ifdef HAVE_ANDROID
 	VPX_CONF += --disable-runtime-cpu-detect
diff --git a/contrib/src/yaml-cpp/rules.mak b/contrib/src/yaml-cpp/rules.mak
index 8ea84c3e5..8b6775929 100644
--- a/contrib/src/yaml-cpp/rules.mak
+++ b/contrib/src/yaml-cpp/rules.mak
@@ -11,7 +11,14 @@ endif
 YAML_CPP_CMAKECONF := -DBUILD_STATIC:BOOL=ON \
                       -DBUILD_SHARED:BOOL=OFF \
                       -DYAML_CPP_BUILD_TOOLS:BOOL=OFF \
-                      -DBUILD_SHARED_LIBS:BOOL=OFF
+                      -DBUILD_SHARED_LIBS:BOOL=OFF \
+                      -DCMAKE_CXX_FLAGS="$(CXXFLAGS)" \
+                      -DCMAKE_CXX_FLAGS_RELEASE="$(CXXFLAGS)" \
+                      -DCMAKE_C_FLAGS="$(CXXFLAGS)" \
+                      -DCMAKE_C_FLAGS_RELEASE="$(CXXFLAGS)" \
+                      -DCMAKE_AR=/usr/bin/gcc-ar \
+                      -DCMAKE_CXX_ARCHIVE_CREATE="<CMAKE_AR> qcs <TARGET> <LINK_FLAGS> <OBJECTS>" \
+                      -DCMAKE_CXX_ARCHIVE_FINISH=true
 
 $(TARBALLS)/yaml-cpp-$(YAML_CPP_VERSION).tar.gz:
 	$(call download,$(YAML_CPP_URL))
@@ -25,5 +32,5 @@ yaml-cpp: yaml-cpp-$(YAML_CPP_VERSION).tar.gz .sum-yaml-cpp
 
 .yaml-cpp: yaml-cpp toolchain.cmake
 	cd $< && $(HOSTVARS) $(CMAKE) . $(YAML_CPP_CMAKECONF)
-	cd $< && $(MAKE) install
+	cd $< && $(MAKE) install VERBOSE=1
 	touch $@
diff --git a/src/media/media_decoder.h b/src/media/media_decoder.h
index 64fd59a24..9fabb007f 100644
--- a/src/media/media_decoder.h
+++ b/src/media/media_decoder.h
@@ -21,6 +21,7 @@
 
 #include "config.h"
 
+#include "libav_deps.h" // MUST BE INCLUDED FIRST
 #include "rational.h"
 
 #ifdef RING_VIDEO
