diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7bb635b..03b96b7 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -94,6 +94,9 @@ FIND_PACKAGE(Qt5 ${QT_MIN_VERSION} CONFIG REQUIRED
   PrintSupport
   QuickControls2
   Quick
+  TextToSpeech
+  X11Extras
+  Concurrent
 )
 
 #FIND_PACKAGE(Qt5TextToSpeech ${QT_MIN_VERSION} QUIET)
@@ -120,6 +123,10 @@ FIND_PACKAGE(KF5 "${KF5_DEP_VERSION}" REQUIRED COMPONENTS
   NotifyConfig
   GlobalAccel
   Declarative
+  TextWidgets
+  ItemViews
+  KIO
+  Archive
 )
 
 FIND_PACKAGE(KF5 "${AKO_DEP_VERSION}" QUIET COMPONENTS
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index a9d9538..2fb464a 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -459,7 +459,7 @@ TARGET_LINK_LIBRARIES(ring-kde
 
 IF (Qt5TextToSpeech_FOUND)
    TARGET_LINK_LIBRARIES(ring-kde
-      Qt5TextToSpeech
+      Qt5::TextToSpeech
    )
 ENDIF()
 
diff --git a/src/accessibility.cpp b/src/accessibility.cpp
index 2d530cc..cd66a52 100644
--- a/src/accessibility.cpp
+++ b/src/accessibility.cpp
@@ -31,7 +31,6 @@
 #include <call.h>
 #include <callmodel.h>
 #include <contactmethod.h>
-#include "kspeechinterfacesingleton.h"
 #include "phonewindow.h"
 #include "view.h"
 
@@ -79,8 +78,8 @@ Accessibility* Accessibility::instance()
 void Accessibility::listCall()
 {
    if (CallModel::instance().getActiveCalls().size()>0) {
-      m_Speech->say(i18np("You currently have <numid>%1</numid> call","You currently have <numid>%1</numid> calls",CallModel::instance()->getActiveCalls().size()));
-      foreach (Call* call,CallModel::instance()->getActiveCalls()) {
+      m_Speech->say(i18np("You currently have <numid>%1</numid> call","You currently have <numid>%1</numid> calls",CallModel::instance().getActiveCalls().size()));
+      foreach (Call* call,CallModel::instance().getActiveCalls()) {
          m_Speech->say(i18n("Call from %1, number %2",call->peerName(),numberToDigit((!call->peerContactMethod()->uri().isEmpty())?call->peerContactMethod()->uri():call->dialNumber())));
       }
    }
@@ -127,7 +126,7 @@ void Accessibility::currentCallDetails()
 ///Helper function is make code shorter
 void Accessibility::say(const QString &message)
 {
-   m_speech->say(message);
+   m_Speech->say(message);
 }
 
 // kate: space-indent on; indent-width 3; replace-tabs on;
diff --git a/src/main.cpp b/src/main.cpp
index 4dc41f4..a5d8418 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -1,3 +1,16 @@
+#include <QLibraryInfo>
+#include <QtPlugin>
+Q_IMPORT_PLUGIN(QXcbIntegrationPlugin)
+Q_IMPORT_PLUGIN(QtQuick2Plugin)
+Q_IMPORT_PLUGIN(QtQuickControls2Plugin)
+Q_IMPORT_PLUGIN(QtQuick2WindowPlugin)
+Q_IMPORT_PLUGIN(QEvdevKeyboardPlugin)
+Q_IMPORT_PLUGIN(QEvdevMousePlugin)
+Q_IMPORT_PLUGIN(QtQuickLayoutsPlugin)
+Q_IMPORT_PLUGIN(QtQuickTemplates2Plugin)
+Q_IMPORT_PLUGIN(QtQuickControls1Plugin)
+Q_IMPORT_PLUGIN(QJpegPlugin)
+Q_IMPORT_PLUGIN(QSvgPlugin)
 /***************************************************************************
  *   Copyright (C) 2009-2015 by Savoir-Faire Linux                         *
  *   Author : Jérémy Quentin <jeremy.quentin@savoirfairelinux.com>         *
@@ -36,15 +49,21 @@ constexpr static const char version[] = "3.0.0-alpha1";
 
 int main(int argc, char **argv)
 {
+#ifdef QT_DEBUG
+  qDebug() << "Running a debug build";
+#else
+  qDebug() << "Running a release build" << QLibraryInfo::isDebugBuild();
+#endif
    try
    {
       //QQmlDebuggingEnabler enabler;
-
+qDebug() << "\n\nBEFORE";
       QGuiApplication::setAttribute(Qt::AA_EnableHighDpiScaling);
 
       RingApplication app( argc, argv );
-
+qDebug() << "\n\nAPP EXIST" << QGuiApplication::instance();
       KLocalizedString::setApplicationDomain("ring-kde");
+qDebug() << "\n\nAPP EXIST2" << QGuiApplication::instance();
 
       KAboutData about(QStringLiteral("ring-kde"),
          i18n("ring-kde"),
@@ -57,6 +76,7 @@ int main(int argc, char **argv)
          QStringLiteral("ring@gnu.org")
       );
       about.setProgramLogo(QImage(QStringLiteral(":appicon/icons/64-apps-ring-kde.png")));
+qDebug() << "\n\nAPP EXIST3" << QGuiApplication::instance();
 
       about.addAuthor( i18n( "Emmanuel Lepage-Vallée"          ), QString(), QStringLiteral("elv1313@gmail.com"                    ));
       about.addAuthor( i18n( "Alexandre Lision"                ), QString(), QStringLiteral("alexandre.lision@savoirfairelinux.com"));
@@ -71,15 +91,6 @@ int main(int argc, char **argv)
       app.setOrganizationDomain(QStringLiteral("ring.cx"));
 
       //Only start the application once
-#ifdef Q_OS_LINUX
-#ifndef DISABLE_KDBUS_SERVICE
-      KDBusService service(KDBusService::Unique);
-      QObject::connect(&service, &KDBusService::activateActionRequested, Cmd::instance(), &Cmd::slotActivateActionRequested);
-      QObject::connect(&service, &KDBusService::activateRequested      , Cmd::instance(), &Cmd::slotActivateRequested      );
-      QObject::connect(&service, &KDBusService::openRequested          , Cmd::instance(), &Cmd::slotOpenRequested          );
-#endif
-#endif
-
       //The app will have quitted by now if an instance already exist
       app.newInstance();
 
