diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3f15aa36..8ead2a2d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -96,6 +96,8 @@ FIND_PACKAGE(Qt5 ${QT_MIN_VERSION} CONFIG REQUIRED
   PrintSupport
   QuickControls2
   Quick
+  Concurrent
+  X11Extras
 )
 
 #FIND_PACKAGE(Qt5TextToSpeech ${QT_MIN_VERSION} QUIET)
@@ -121,7 +123,10 @@ FIND_PACKAGE(KF5 "${KF5_DEP_VERSION}" REQUIRED COMPONENTS
   NotifyConfig
   GlobalAccel
   Declarative
-  Kirigami2
+  #Kirigami2
+  ItemViews
+  Archive
+
 )
 
 FIND_PACKAGE(KF5 "${AKO_DEP_VERSION}" QUIET COMPONENTS
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index e48fe41e..438e56ef 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -459,7 +459,6 @@ TARGET_LINK_LIBRARIES(ring-kde
   KF5::NotifyConfig
   KF5::GlobalAccel
   KF5::Declarative
-  KF5::Kirigami2
 )
 
 IF (Qt5TextToSpeech_FOUND)
diff --git a/src/timeline/CMakeLists.txt b/src/timeline/CMakeLists.txt
index 3e32cf3d..120232d0 100644
--- a/src/timeline/CMakeLists.txt
+++ b/src/timeline/CMakeLists.txt
@@ -79,7 +79,7 @@ target_link_libraries( libtimeline
    KF5::WidgetsAddons
    KF5::ConfigCore
    KF5::ConfigGui
-   KF5::Kirigami2
+   #KF5::Kirigami2
 )
 
 SET( libtimeline_LIB_HDRS
